scripts/migrations/2020-08-update_migration_created_date.py

```diff
@@ -0,0 +1,118 @@
+dynamodb = get_aws_resource("dynamodb", role_arn=READ_ONLY_ROLE_ARN)
+s3 = get_aws_resource("s3", role_arn=READ_ONLY_ROLE_ARN)
+
+vhs_table = "vhs-storage-staging-manifests-2020-07-24"
+
+backfill_vhs_table = "vhs-storage-staging-manifests-2020-08-19"
+
+errors = {}
+
+
+def record_error(id, version, err):
+    print(f"\033[91mError while updating {id}/{version}: {err}")
+    errors[f'{id}/{version}'] = err
+
+def get_bucket_key(item):
+    try:
+        bucket = item["payload"]["namespace"]
+        key = item["payload"]["path"]
+    except KeyError:
+        try:
+            bucket = item["payload"]["bucket"]
+            key = item["payload"]["key"]
+        except:
+            record_error(
+                item["id"], item["version"], f"Cannot find s3 bucket, key in {item}"
+            )
+        else:
+            return bucket, key
+    else:
+        return bucket, key
+
+
+def get_vhs_json(id, version, bucket, key):
+    try:
+        return json.loads(s3.Object(bucket, key).get()["Body"].read().decode("utf-8"))
+    except ClientError as e:
+        record_error(id, version, f"Cannot read s3 entry for {id}: {item}: {e}")
+
+
+def put_vhs_json(id, version, bucket, item, content):
+    filename = f"{uuid.uuid4()}.json"
+    key = f"{id}/{version}/{filename}"
+    item["payload"]["bucket"] = bucket
+    item["payload"]["key"] = key
+    try:
+        s3.Object(bucket, key).put(Body=(bytes(json.dumps(content).encode("UTF-8"))))
+        dynamodb.Table(vhs_table).put_item(Item=item)
+    except ClientError as e:
+        record_error(
+            id,
+            version,
+            f"Error updating backfill vhs object {id}/{version} to s3://{bucket}/{key}: {e}",
+        )
+
+
+def get_backfill_item(id, version):
+    try:
+        return dynamodb.Table(backfill_vhs_table).get_item(
+            Key={"id": id, "version": version}
+        )["Item"]
+    except KeyError:
+        record_error(id, version, f"Cannot find backfill storage manifest!!!!")
+
+
+def is_expected_diff(id, version, current_json, backfill_json):
+    diff = DeepDiff(current_json, backfill_json, ignore_order=True)
+
+    if diff:
+        values_changed = diff.pop("values_changed", None)
+        items_added = diff.pop("iterable_item_added", None)
+        if values_changed and values_changed.keys() - ["root['createdDate']"]:
+            record_error(id, version, f"Unexpected values changed in {values_changed}!")
+            return False
+        if items_added and items_added.keys() - ["root['replicaLocations'][1]"]:
+            record_error(id, version, f"Unexpected values changed in {items_added}!")
+            return False
+        if diff:
+            record_error(id, version, f"Unexpected values changed in {diff}!")
+            return False
+    return True
+
+
+for item in scan_table(TableName=vhs_table):
+    id = item["id"]
+    version = item["version"]
+    bucket, key = get_bucket_key(item)
+    if bucket and key:
```

Comment:
So FWIW, I suspect this would explode if `get_bucket_key()` ever failed to find anything. It would return `None`, and this line would become

```python
bucket, key = None
```

which is a TypeError – you can’t unpack `None` into two variables.

I doubt it'll be an issue here, but a note for the next Python you write.